### ISSUE TITLE
`set -g extended-keys always` breaks <S-Tab>

### ISSUE THREAD

**Issue by kjuq** [tmux/tmux#4304]
### Issue description

`set -g extended-keys always` breaks \<S-Tab\>

#### Minimal reproduction.

```bash
echo 'set -g extended-keys always' > /tmp/tmux.conf
tmux -f /tmp/tmux.conf
```

Then press \<S-Tab\> on bash or fish inside tmux, a raw sequence will be shown such as `[27;2;9~` and `;2;9~`.

![bash](https://github.com/user-attachments/assets/f766b9ea-5b01-4db5-b66c-daf811e66117)
![fish](https://github.com/user-attachments/assets/8c344c6f-320c-4d96-8d43-12926e23db35)

### Required information

- tmux version (`tmux -V`).
	- `tmux next-3.6`
- Platform (`uname -sp`).
	- `Linux unknown` (Manjaro)
- $TERM inside and outside of tmux (`echo $TERM`).
	- Inside: `tmux-256color`
	- Outside: `alacritty`
- Logs from tmux (`tmux kill-server; tmux -vv new`).

**Comment by nicm** [comment]
I do not see you pressing S-Tab in this log, only C-Tab (S-Tab would be 2 not 5):
~~~~
1734544808.320693 /dev/pts/0: keys are 9 (\033[27;5;9~)
1734544808.320702 /dev/pts/0: extended key \033[27;5;9~ is 200000000009 (C-Tab)
1734544808.320706 /dev/pts/0: complete key \033[27;5;9~ 0x200000000009
~~~~
What terminal are you using? tmux is not asking for extended keys so it appears it is sending this format unprompted.

If you run `cat` *outside tmux* and then press `S-Tab`, what do you see?

**Comment by kjuq** [comment]
Thank you for your reply!

The issue was reproduced with Alacritty and Wezterm as far as I tried.

- `alacritty --config-file /dev/null`
	- `alacritty 0.14.0 (ecf27800)`
- `wezterm --skip-config`
	- `wezterm 20240203-110809-5046fc22`

And the output from `cat` with pressing `<S-Tab>` outside of tmux is `^[[Z`.
FYI, the output inside of tmux is `^[[27;2;9~` of course.

**Comment by nicm** [comment]
So it is working? You are meant to get `^[[27;2;9~` for `S-Tab` in tmux with extended keys...

**Comment by kjuq** [comment]
I think it is not working properly? Because I realized that I can move back and forth at the fish completion with `tmux 3.4` but not with `tmux 3.5a`.

The videos below are recordings of each behaviors.

- `tmux 3.5a`
	- https://github.com/user-attachments/assets/34b5a048-7302-4317-82bb-c0e17380d4a9
- `tmux 3.4`
	- https://github.com/user-attachments/assets/8186aef9-f8f2-42fd-9d1d-3b123b6c1a43

**Comment by nicm** [comment]
Show server log of you pressing S-Tab please, although I can only tell you whether tmux is sending the right thing, not what the applications do with it.

**Comment by kjuq** [comment]
Sure. I dumped log using `tmux kill-server; tmux -vv new`. On each sessions, I pressed `<Tab>` and `<S-Tab>` several times.

- `3.4`: [tmux-server-325514.log](https://github.com/user-attachments/files/18196697/tmux-server-325514.log)
- `3.5a`: [tmux-server-326567.log](https://github.com/user-attachments/files/18196696/tmux-server-326567.log)

**Comment by nicm** [comment]
Yes I think tmux should still be sending `\033[Z` for `S-Tab` even with extended keys, I'll take a look later.

**Comment by kjuq** [comment]
Thank you! I'm looking forward to your inspection

**Comment by nicm** [comment]
Looks like we should send `\033[Z` in mode 1 and `^[[27;2;9~` only in mode 2.

Does this fix it? [x.diff.txt](https://github.com/user-attachments/files/18198903/x.diff.txt)

**Comment by kjuq** [comment]
Thank you very much!! It seems to be fixed. I'm so sorry for not providing the proper logs initially. It was helpful for your support.

**Comment by nicm** [comment]
Great, applied to OpenBSD now, will be in GitHub later. Thanks!
